src/backend/src/routers/signup.js

```diff
@@ -65,273 +65,281 @@ module.exports = eggspress(['/signup'], {
         body: req.body,
     });
 
-    // check bot trap, if `p102xyzname` is anything but an empty string it means
-    // that a bot has filled the form
-    // doesn't apply to temp users
-    if(!req.body.is_temp && req.body.p102xyzname !== '')
-        return res.send();
-
-    // check if user is already logged in
-    if ( req.body.is_temp && req.cookies[config.cookie_name] ) {
-        const { user, token } = await svc_auth.check_session(
-            req.cookies[config.cookie_name]
-        );
-        res.cookie(config.cookie_name, token, {
-            sameSite: 'none',
-            secure: true,
-            httpOnly: true,
-        });
-        // const decoded = await jwt.verify(token, config.jwt_secret);
-        // const user = await get_user({ uuid: decoded.uuid });
-        if ( user ) {
-            return res.send({
-                token: token,
-                user: {
-                    username: user.username,
-                    uuid: user.uuid,
-                    email: user.email,
-                    email_confirmed: user.email_confirmed,
-                    requires_email_confirmation: user.requires_email_confirmation,
-                    is_temp: (user.password === null && user.email === null),
-                    taskbar_items: await get_taskbar_items(user),
-                }
+    if(config.registration_closed==true)
+    {
+        return res.status(403).send('Registration is closed.');
+    }
+    else
```

Comment:
We don't need an `else` if there's an early return. That will reduce indentation changes and make this easier to review.

Reply:
updated